api/types/swarm/secret.go

```diff
@@ -0,0 +1,33 @@
+package swarm
+
+import "os"
+
+// Secret represents a secret.
+type Secret struct {
+	ID string
+	Meta
+	Spec       SecretSpec
+	Digest     string
+	SecretSize int64
+}
+
+// SecretSpec represents a secret specification from a secret in swarm
+type SecretSpec struct {
+	Annotations
+	Data []byte `json:",omitempty"`
+}
+
+// SecretReferenceFileTarget is a file target in a secret reference
+type SecretReferenceFileTarget struct {
+	Name string
+	UID  string
+	GID  string
+	Mode os.FileMode
+}
+
+// SecretReference is a reference to a secret in swarm
+type SecretReference struct {
+	SecretID   string
+	SecretName string
+	Target     *SecretReferenceFileTarget
```

Comment:
@ehazlett That, or just pull all this up to the main reference type:

```
type SecretReference {
   // ...
  *File, *Env
}
```

This is more consistent with what we did for runtime, I believe.